lib/elixir/lib/inspect/algebra.ex

```diff
@@ -88,8 +94,30 @@ defmodule Inspect.Opts do
           pretty: boolean,
           safe: boolean,
           syntax_colors: [{color_key, IO.ANSI.ansidata()}],
-          inspect_fun: (any, t -> Inspect.Algebra.t())
+          inspect_fun: (any, t -> Inspect.Algebra.t()),
+          custom_options: keyword
         }
+
+  @doc """
+  Filters out all the default options, returning the custom options only.
+  """
+  @spec filter_custom_options(keyword()) :: keyword()
```

Comment:
sorry for the noise, but I think this one reads better
```suggestion
  @spec filter_custom_options(opts) :: opts when opts: keyword()
```

Reply:
No worries, pushed.